kolibri/plugins/management/assets/src/vue/manage-class-page/class-delete-modal.vue

```diff
@@ -68,14 +62,10 @@
         required: true,
       },
     },
-    data() {
-      return {
-        error_message: '',
-      };
-    },
     methods: {
-      submit() {
-        // need delete logic here.
+      classDelete() {
+        this.deleteClass(this.classid);
+        this.$emit('close');
```

Comment:
I thought you oppose the idea to have action directly control the view via promise.
And I don't see how this solution solves mutation bloating.

I like the idea to set more informative value to `pageState.modalShown`.
So we have all modals react to the same `modalShown`
when `modalShown` is set to null or 0, all modals hide
when `modalShown` is set to `classNameEdit` string, only class-name-edit modal shown.
So the view is controlled by vuex store, not the action.
It's pretty much a event pattern implemented via vuex. Thought?